data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/source/mongoDB/MongoDBSnapshotWorker.java

```diff
@@ -0,0 +1,207 @@
+                    return;
+                }
+            }
+        }
+    }
+
+    private void startProcessPartition(SourcePartition<MongoDBSnapshotProgressState> partition) {
+        List<String> partitionKeys = List.of(partition.getPartitionKey().split("\\|"));
+        if (partitionKeys.size() < 4) {
+            throw new RuntimeException("Invalid Partition Key. Must as db.collection|gte|lte format.");
+        }
+        List<String> collection = List.of(partitionKeys.get(0).split("\\."));
+        final String gte = partitionKeys.get(1);
+        final String lte = partitionKeys.get(2);
+        final String className = partitionKeys.get(3);
+        if (collection.size() < 2) {
+            throw new RuntimeException("Invalid Collection Name. Must as db.collection format");
+        }
+        try (MongoClient mongoClient = MongoDBHelper.getMongoClient(mongoDBConfig)) {
+            MongoDatabase db = mongoClient.getDatabase(collection.get(0));
+            MongoCollection<Document> col = db.getCollection(collection.get(1));
+            Bson query = MongoDBHelper.buildAndQuery(gte, lte, className);
+            long totalRecords = 0L;
+            long successRecords = 0L;
+            long failedRecords = 0L;
+            try (MongoCursor<Document> cursor = col.find(query).iterator()) {
+                while (cursor.hasNext()) {
+                    totalRecords += 1;
+                    try {
+                        JsonWriterSettings writerSettings = JsonWriterSettings.builder()
+                                .outputMode(JsonMode.RELAXED)
+                                .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                                .build();
+                        String record = cursor.next().toJson(writerSettings);
+                        Map<String, Object> data = convertToMap(record);
+                        data.putIfAbsent(EVENT_SOURCE_DB_ATTRIBUTE, collection.get(0));
+                        data.putIfAbsent(EVENT_SOURCE_COLLECTION_ATTRIBUTE, collection.get(1));
+                        data.putIfAbsent(EVENT_SOURCE_OPERATION, OpenSearchBulkActions.CREATE.toString());
+                        data.putIfAbsent(EVENT_SOURCE_TS_MS, 0);
+                        if (buffer.isByteBuffer()) {
+                            buffer.writeBytes(objectMapper.writeValueAsBytes(data), null, DEFAULT_BUFFER_WRITE_TIMEOUT_MS);
+                        } else {
+                            buffer.write(getEventFromData(data), DEFAULT_BUFFER_WRITE_TIMEOUT_MS);
+                        }
+                        successItemsCounter.increment();
+                        successRecords += 1;
+                    } catch (Exception e) {
+                        LOG.error("failed to add record to buffer with error {}", e.getMessage());
+                        failureItemsCounter.increment();
```

Comment:
will these records be retries by acknowledgementSet manager ?

Reply:
From my understanding, it will. But please double confirm with Tylor.
I am writing same as other places where acknowledgementSet is used.